### PR TITLE
feat: #263 Added i18n,  error interface library

### PR DESCRIPTION
**Pull request by impiyush83**
**What kind of change does this PR introduce?**
#263 
Standardized class for throwing error with i18n internationalization

**Did you add tests for your changes?**
N/A

**If relevant, did you update the documentation?**
N/A

**Summary**
Helps developers to throw errors supporting i18n 

**Does this PR introduce a breaking change?**
N/A

**Other information**

### CODE REVIEW

**Comment by impiyush83** [comment]
@DMills27 @Laurell876  please review the PR

**Comment by palisadoes** [comment]
@sumitra19jha Please take a look at this as I know you are working on multilingual support.

**Comment by sumitra19jha** [comment]
@impiyush83 Can you send a screen recording of API testing?

**Comment by impiyush83** [comment]
@sumitra19jha  

Currently, we have defaultLocale set as 'en'. 

We can introduce toggling of locales before translating depending on the #280  language code

https://drive.google.com/file/d/1ID9sH-pd4TtfOV3vwLVEIs4apbkczKxH/view?usp=sharing

File: resolvers/Query.js
Comment (sumitra19jha) [inline resolvers/Query.js:100]: Can you explain what is this translate doing?

File: core/errors/index.js
Comment (sumitra19jha) [inline core/errors/index.js:3]: Can you please add the comments on how each error are different from each other?
This will give more clarity on something like how is `ConflictError` different from `NotFoundError`

File: core/libs/talawa-request-context.js
Comment (sumitra19jha) [inline core/libs/talawa-request-context.js:46]: There is no as such json file for error message translation. Can you add a demo of how to throw error in some other language like French, Hindi or Spanish

File: resolvers/Query.js
Comment (sumitra19jha) [reply resolvers/Query.js:100]: And What about custom message translation?
Let say i want to throw error: `User access is denied`, so how we will deal with that custom translation?

File: core/errors/index.js
Comment (impiyush83) [reply core/errors/index.js:3]: They mostly differ by HTTP code like,

Conflict - 409
Validation - 422 
etc 

Refer: https://www.restapitutorial.com/httpstatuscodes.html

**Comment by impiyush83** [comment]
https://drive.google.com/file/d/1TJJa9hBldAtHPbZYoLGC9Be5JUqzp4Xe/view?usp=sharing

Updated error response

**Comment by palisadoes** [comment]
@impiyush83  @sumitra19jha What's the status on this? It doesn't sound like it's ready for review.

**Comment by sumitra19jha** [comment]
I didn't see the suggestion being updated.
The comments for distinguishing error is missing and it's not clear what is the role of translate. If its responsible for translating error message, then I request to attach a snapshot

File: core/errors/index.js
Comment (sumitra19jha) [reply core/errors/index.js:3]: Can you please add the comment as when to use which

**Comment by palisadoes** [comment]
> I didn't see the suggestion being updated.
> The comments for distinguishing error is missing and it's not clear what is the role of translate. If its responsible for translating error message, then I request to attach a snapshot

@impiyush83  Please respond.

File: core/errors/index.js
Comment (impiyush83) [reply core/errors/index.js:3]: @sumitra19jha  @palisadoes  

1. Validation Error 

Suppose you are validating some things like :
- Check if the file Type is in 'png' if not throw Validation Error 
- Validation error has the capability to throw multiple validation errors at once with 422 HTTP Code

2. Conflict Error 

 - Suppose an entry is already present in DB which ideally should be immutable
 - eg: User of id 1, New user should not be added again with id 1 here it should raise Conflict error with 409 HTTP Code

3. NotFound Error 

- Anything entry which you expect to exist but not found here you will throw NotFoundError

4. UnAuthorizedError 

- Who has authentication but is Forbidden to access a resource here we will send this with 403 HTTP Code

5. UnAuthenticatedError 

- Invalid Credentials, invalid auth token 
- Here we will send UnAuthenticatedError with 401 HTTP Code

6. ApplicationError 

- ApplicationError is like interface here.

7.  InternalServerError 

- Should be thrown if we don't know the type of the error which is a very rare use case 
- eg: cant start server, db connection error. 

These are all the errors added :

errors.ApplicationError = require('./application-error');
errors.InternalServerError = require('./internal-server-error');
errors.NotFoundError = require('./not-found-error');
errors.UnauthorizedError = require('./unauthorized-error');
errors.ValidationError = require('./validation-error');
errors.UnauthenticatedError = require('./unauthenticated-error');
errors.ConflictError = require('./conflict-error');

File: index.js
Comment (sumitra19jha) [reply index.js:55]: Can you add snapshot of how if we want we can change the language?

File: locales/en.json
Comment (sumitra19jha) [reply locales/en.json:1]: I can see only one language. Currently we support 5 languages. (en, fr, sp, zh, hi)

Comment (sumitra19jha) [reply locales/en.json:1]: Please add json files for these languages

**Comment by impiyush83** [comment]
@palisadoes  @sumitra19jha  Added video to show how we can toggle languages via Accept-Language header.


https://user-images.githubusercontent.com/20107021/119901485-9a84e400-bf63-11eb-9776-772588b0d47d.mov

**Comment by palisadoes** [comment]
@impiyush83  I like the way the `es` edge case is handled. I'll wait for @sumitra19jha to respond. 

@DMills27 is waiting for the outcome before doing the technical review.

**Comment by impiyush83** [comment]
@sumitra19jha  @palisadoes  

All supported languages video ['hi', 'en', 'zh', 'fr', 'sp']

https://user-images.githubusercontent.com/20107021/119906739-63670080-bf6c-11eb-9cb2-785b69825be4.mov

**Review by Laurell876 — APPROVED** [review]

**Review by palisadoes — CHANGES_REQUESTED** [review]
@sumitra19jha does this match your plans for multilingual support? Please confirm before I merge

**Review by sumitra19jha — APPROVED** [review]
@palisadoes This PR contains 82 files being changed with none done for test cases. So Travis checks might fail in Future.

**Pull request merged**
